src/Stack/Types/Config.hs

```diff
@@ -365,6 +365,10 @@ data Resolver
   -- ^ Require a specific GHC major version, but otherwise provide no build
   -- plan. Intended for use cases where end user wishes to specify all upstream
   -- dependencies manually, such as using a dependency solver.
+
+  | ResolverCustom !Text
+  -- ^ A custom resolver based on the given URL. This file is assumed to be
```

Comment:
For local files, easily. It's for remote files that we have a problem since it would be inefficient to download every time. This actually seems necessary to handle relative files properly...